internal/provider/private_endpoint_services_resource.go

```diff
@@ -302,6 +315,11 @@ func waitForEndpointServicesCreatedFunc(
 				return retry.RetryableError(fmt.Errorf("encountered a server error while reading endpoint status - trying again"))
```

Comment:
It seems like we've lost the error here. Not inspecting it or logging it anywhere seems error prone.  Can we not at least show it to the user in this case?

Reply:
done.